### PR TITLE
Installed firmware verification tool

### PR DESCRIPTION
**Pull request by AlCutter**
TODO: needs #35 and #37 in first

This PR adds a new `verify` command which is intended to be used by device custodians to check that the pre-installed "factory" firmware is all present in the firmware transparency log.

Example run against a correctly flashed dev device:
```bash
./verify --binaries_url=file://${DEV_LOG_DIR}/ --firmware_log_url=file://${DEV_LOG_DIR}/log/ --firmware_log_origin="DEV.armoredwitness.transparency.dev/${USER}" --firmware_log_verifier=$(cat ${LOG_PUBLIC_KEY}) --applet_verifier=$(cat ${APPLET_PUBLIC_KEY}) --boot_verifier=$(cat ${BOOT_PUBLIC_KEY}) --recovery_verifier=$(cat ${RECOVERY_PUBLIC_KEY}) --os_verifier_1=$(cat ${OS_PUBLIC_KEY1}) --os_verifier_2=$(cat ${OS_PUBLIC_KEY2})
I1020 16:21:58.335172  564090 main.go:152] Fetching RECOVERY bin from "recovery/0.0.0-incompatible+v20231018/armory-ums.imx"
I1020 16:21:58.338765  564090 main.go:204] Successfully fetched and verified recovery image
I1020 16:21:58.338771  564090 main.go:205] ----------------------------------------------------------------------------------------------
I1020 16:21:58.338773  564090 main.go:206] 🙏 Operator, please ensure boot switch is set to USB, and then connect unprovisioned device 🙏
I1020 16:21:58.338777  564090 main.go:207] ----------------------------------------------------------------------------------------------
I1020 16:21:58.338802  564090 recovery.go:60] Waiting for device to be detected...
I1020 16:21:59.396450  564090 sdp.go:85] found device 15a2:007d Freescale SemiConductor Inc  SE Blank 6UL
I1020 16:21:59.460026  564090 sdp.go:111] Attempting to SDP boot device /dev/hidraw0
I1020 16:21:59.460104  564090 sdp.go:123] Loading DCD at 0x00910000 (976 bytes)
I1020 16:21:59.463333  564090 sdp.go:128] Loading imx to 0x8000f400 (1793024 bytes)
I1020 16:22:00.427265  564090 sdp.go:133] Sending jump address to 0x8000f400
I1020 16:22:00.427512  564090 sdp.go:138] Serial download on /dev/hidraw0 complete
I1020 16:22:00.427667  564090 recovery.go:47] Witness device booting recovery image
I1020 16:22:00.427684  564090 recovery.go:102] Waiting for block device to appear
I1020 16:22:02.241549  564090 main.go:215] ✅ Detected device "/dev/hidraw0"
I1020 16:22:02.241591  564090 main.go:216] ✅ Detected blockdevice /dev/sdc
W1020 16:22:02.241678  564090 main.go:224] Failed to extract firmware: error opening /dev/sdc: open /dev/sdc: no such device or address
I1020 16:22:03.241948  564090 main.go:226] Retrying...
I1020 16:22:03.247033  564090 main.go:352] Found config at block 0x4fb0
I1020 16:22:03.247567  564090 main.go:353] Reading 0x2afc00 bytes of firmware from MMC byte offset 0x400
I1020 16:22:03.379725  564090 main.go:352] Found config at block 0x5000
I1020 16:22:03.379933  564090 main.go:353] Reading 0x93d000 bytes of firmware from MMC byte offset 0xa0a000
I1020 16:22:03.820183  564090 main.go:352] Found config at block 0x200000
I1020 16:22:03.820229  564090 main.go:353] Reading 0xe95000 bytes of firmware from MMC byte offset 0x4000a000
I1020 16:22:04.531375  564090 main.go:273]   ✅ Bootloader: proof bundle is self-consistent
I1020 16:22:04.531622  564090 main.go:296]   ✅ Bootloader: proof bundle checkpoint(@4) is consistent with current view of log(@4)
I1020 16:22:04.544748  564090 main.go:273]   ✅ TrustedOS: proof bundle is self-consistent
I1020 16:22:04.544822  564090 main.go:296]   ✅ TrustedOS: proof bundle checkpoint(@4) is consistent with current view of log(@4)
I1020 16:22:04.554906  564090 main.go:273]   ✅ TrustedApplet: proof bundle is self-consistent
I1020 16:22:04.554975  564090 main.go:296]   ✅ TrustedApplet: proof bundle checkpoint(@4) is consistent with current view of log(@4)
I1020 16:22:04.554993  564090 main.go:105] ✅ Device verified OK!
I1020 16:22:04.554996  564090 main.go:106] ----------------------------------------------------------------------------------------------
I1020 16:22:04.555000  564090 main.go:107] 🙏 Operator, please ensure boot switch is set to MMC, and then reboot device 🙏
I1020 16:22:04.555002  564090 main.go:108] ----------------------------------------------------------------------------------------------
```

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -13,11 +13,11 @@ jobs:
     steps:
       - uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
         with:
-          go-version: 1.19
+          go-version: 1.21
```
Comment (mhutchinson) [inline .github/workflows/golangci-lint.yml:16]: I think this should be bumped to 1.20? Also, this and the above change to go_test.yml should be in a separate PR.

**Review by mhutchinson — COMMENTED** [review]
This PR looks like it has a number of misc changes in it. Can you give it a tidy and let me know when it's ready for review?

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (AlCutter) [reply .github/workflows/golangci-lint.yml:16]: I'll move them to another PR no problem, but they're in here because the workflows failed, probably mainly because of `errors.Join`.

Comment (AlCutter) [reply .github/workflows/golangci-lint.yml]: Done in #38

**Comment by AlCutter** [comment]
It might be worth reviewing commit-by-commit on this one since there's a fair chunk of code just being moved around as-is in places.

File: internal/device/recovery.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	"github.com/fsnotify/fsnotify"
+	"k8s.io/klog/v2"
+)
+
+// BootIntoRecovery attempts to boot an armored witness device in SDP (Serial Boot) mode
+// with the recovery firmware image, and then watches for a matching new block device
+// to be presented to the host.
+//
+// Returns the HID device and detected block device path.
+func BootIntoRecovery(ctx context.Context, recoveryFirmware []byte, blockDeviceGlob string) (*Target, string, error) {
```
Comment (mhutchinson) [inline internal/device/recovery.go:34]: If I only saw the function description for this I would assume it would be more active and "do it or fail fast". Specifically, I wouldn't have expected a wait on the context to be the first thing.

Consider `AwaitDeviceAndBootIntoRecovery`, or just line in the docstring making it clear that the context must have a timeout if you don't want this to hang forever if no device appears in the correct state.

**Review by mhutchinson — APPROVED** [review]

File: internal/device/recovery.go (hunk 2 above)
Comment (AlCutter) [reply internal/device/recovery.go]: I've updated the comment on the function to make it clearer, ta.
